shell/platform/fuchsia/dart-pkg/zircon/lib/src/zd_handle.dart

```diff
@@ -0,0 +1,76 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of zircon;
+
+@pragma('vm:entry-point')
+class ZDHandle {
+  @pragma('vm:entry-point')
+  ZDHandle._(this._ptr) {
+    _attachFinalizer();
+  }
+
+  void _attachFinalizer() {
+    // TODO (kaushikiska): fix external allocation size.
+    final int? ret = zirconFFIBindings?.zircon_dart_handle_attach_finalizer(
+        this, _ptr.cast(), 128);
+    if (ret != 1) {
+      throw Exception('Unable to attach finalizer to handle');
+    }
+  }
+
+  int get handle => _ptr.ref.handle;
+
+  final Pointer<zircon_dart_handle_t> _ptr;
+
+  bool isValid() {
+    int? ret = zirconFFIBindings?.zircon_dart_handle_is_valid(_ptr);
+    return ret == 1;
+  }
+
+  bool close() {
+    if (isValid()) {
+      int? ret = zirconFFIBindings?.zircon_dart_handle_close(_ptr);
```

Comment:
Let's assert(isValid()) here, then do the if check.  99% of the time, isValid() being false here indicates a coding error in the dart application

This is the behavior that the existing implementation has (DCHECK on handle validity, then proceed forward with an if)